.helm/templates/deployment.yaml

```diff
@@ -0,0 +1,184 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "app.name" . }}
+  labels:
+    {{- include "app.labels" $ | nindent 4 }}
+  {{- with .Values.additionalAnnotations }}
+  annotations:
+    {{- toYaml . | nindent 4 }}
+  {{- end }}
+spec:
+  selector:
+    matchLabels:
+      {{- include "app.selectorLabels" $ | nindent 6 }}
+  replicas: {{ .Values.scheduler.replicas | default 1 }}
+  # The deployment strategy is hardcoded as well until NCC supports HA mode
+  strategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        {{- include "app.labels" $ | nindent 8 }}
+      {{- with .Values.additionalAnnotations }}
+      annotations:
+        deployment/date: {{ now }} # Force redeployment
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "app.serviceAccountName" . }}
+      {{- with .Values.securityContext }}
+      securityContext:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      containers:
+        - name: nexus
+          args:
+            - "log-level"
+            - {{ .Values.logging.level }}
+        {{- with .Values.securityContext }}
+          securityContext:
+          {{- toYaml . | nindent 12 }}
+        {{- end }}
+          image: "{{ include "app.image" . }}"
+          imagePullPolicy: "{{ .Values.image.pullPolicy }}"
+          env:
+            - name: APPLICATION_VERSION
+              value: "{{ (default (printf "v%s" .Chart.AppVersion) .Values.image.tag) }}"
+            - name: APPLICATION_ENVIRONMENT
+              value: "{{ lower .Values.environment }}"
+            - name: NEXUS__RESOURCE_NAMESPACE
+              value: {{ .Values.scheduler.namespace }}
+            - name: NEXUS__BUFFER.WORKERS
+              value: {{ .Values.buffer.workers | quote }}
+            - name: NEXUS__BUFFER.FAILURE_RATE_BASE_DELAY
+              value: {{ .Values.buffer.backOffBaseDelay }}
+            - name: NEXUS__BUFFER.FAILURE_RATE_MAX_DELAY
+              value: {{ .Values.buffer.backOffMaxDelay }}
+            - name: NEXUS__BUFFER.RATE_LIMIT_ELEMENTS_PER_SECOND
+              value: {{ .Values.buffer.rateLimitElementsPerSecond | quote }}
+            - name: NEXUS__BUFFER.RATE_LIMIT_ELEMENTS_BURST
+              value: {{ .Values.buffer.rateLimitElementsBurst | quote }}
+          {{- if .Values.buffer.s3.enabled }}
+            - name: NEXUS__BUFFER.BUFFER_CONFIG.PAYLOAD_STORAGE_PATH
+              value: {{ .Values.buffer.s3.storagePath }}
+            - name: NEXUS__BUFFER.ACCESS_KEY_ID
+              valueFrom:
+                secretKeyRef:
+                  name: {{ .Values.buffer.s3.connectionSecretName }}
+                  key: accessKeyId
+            - name: NEXUS__BUFFER.SECRET_ACCESS_KEY
```

Comment:
Dots in environment variables - bad idea.

Reply:
This is default separator in Viper. Not sure how to override this